Client/Frontend/Browser/URIFixup.swift

```diff
@@ -34,4 +39,29 @@ class URIFixup {
 
         return nil
     }
-}
\ No newline at end of file
+
+    static func strToAscii(string : String) -> String {
+        var host : String?
+
+        if string.hasPrefix("http://") {
+            host = string.substringFromIndex(string.startIndex.advancedBy(7))
+        } else if string.hasPrefix("https://") {
+            host = string.substringFromIndex(string.startIndex.advancedBy(8))
+        } else {
+            host = string
+        }
+
+        let t = host!.rangeOfString("/")
+        if (t?.startIndex) != nil {
+            host = host?.substringToIndex((t?.startIndex)!)
+        }
+
+        let s = host!.rangeOfString(":")
+        if (s?.startIndex) != nil {
+            host = host?.substringToIndex((t?.startIndex)!)
+        }
+
+        let urlStr = host?.utf8HostToAscii()
+        return urlStr!
```

Comment:
> For puny code string use NSURLComponents(string: string) will return nil, can’t get the host. So we just manually analyze it.

Did you try `punycodedURL(escaped)`, not `punycodedURL(trimmed)`? From my testing, using the escaped string produces a valid `NSURL`.